SimTracker/TrackTriggerAssociation/plugins/TTStubAssociator.cc

```diff
@@ -64,36 +60,28 @@ void TTStubAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, const
 
         DetId stackDetid = tTopo->stack(detid);  // Stub module detid
 
-        if (TTStubHandle->find(stackDetid) == TTStubHandle->end())
+        if (ttStubHandle->find(stackDetid) == ttStubHandle->end())
           continue;
 
         /// Get the DetSets of the Clusters
-        edmNew::DetSet<TTStub<Ref_Phase2TrackerDigi_>> stubs = (*TTStubHandle)[stackDetid];
+        edmNew::DetSet<TTStub<Ref_Phase2TrackerDigi_>> stubs = (*ttStubHandle)[stackDetid];
 
         for (auto contentIter = stubs.begin(); contentIter != stubs.end(); ++contentIter) {
           /// Make the reference to be put in the map
-          edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>, TTStub<Ref_Phase2TrackerDigi_>> tempStubRef =
-              edmNew::makeRefTo(TTStubHandle, contentIter);
-
-          /// Get the two clusters
-          //        std::vector< edm::Ref< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > >, TTCluster< Ref_Phase2TrackerDigi_ > > > theseClusters = tempStubRef->getClusterRefs();
+          TTStubRef tempStubRef = edmNew::makeRefTo(ttStubHandle, contentIter);
 
           /// Fill the inclusive map which is careless of the stub classification
           for (unsigned int ic = 0; ic < 2; ic++) {
-            std::vector<edm::Ptr<TrackingParticle>> tempTPs =
-                TTClusterAssociationMapHandle->findTrackingParticlePtrs(tempStubRef->clusterRef(ic));
-
-            for (unsigned int itp = 0; itp < tempTPs.size(); itp++) {
-              edm::Ptr<TrackingParticle> testTP = tempTPs.at(itp);
+            std::vector<TrackingParticlePtr> tempTPs =
```

Comment:
capture by const ref

Reply:
Can't. Implementation of TTClusterAssociationMap::findTrackingParticlePtrs(cluster) returns a temporary vector if cluster is not present in the map.

Reply:
a) my review includes a proposal to improve that behavior
b) you can capture a temporary return value by const ref to extend its lifetime

Reply:
agreed. done